lib/Value/Matrix.pm

```diff
@@ -720,26 +906,43 @@ sub Zero {
 	return $self->make($context, @M);
 }
 
-#
-#  Extract a given row from the matrix
-#
+=head3 C<row>
+
+Extract a given row from the matrix.
+
+Usage:
+
+    my $A1 = Matrix([ [ 1, 2, 3, 4 ], [ 5, 6, 7, 8 ], [ 9, 10, 11, 12 ] ]);
+    $A1->row(2);  # returns the row Matrix [5,6,7,8]
+
+=cut
+
 sub row {
 	my $self = (ref($_[0]) ? $_[0] : shift);
 	my $M    = $self->promote(shift);
 	my $i    = shift;
-	return if $i == 0;
-	$i--   if $i > 0;
+	Value::Error("Row must be a positive integer") unless $i =~ m/^[1-9]\d*$/;
```

Comment:
I can't recall where, but there is somewhere else like this where we support having an explicit plus sign in front of the integer.

Reply:
But maybe this would be better if if actually accessed the dimensions of the matrix, and required an integer that is a valid row index.

Reply:
Since `$i` is usually a Perl real (not a string, for example), an initial plus sign will not be in its stringified version, so won't be part of the comparison in `$i =~ m/.../` even if you used `$M->row(+3)`.  It would only be there if you passed the row number as a string, as in `$M->row("+3")`, which I might consider to be an improper value to pass (it is *not* a positive integer, it is a string).